src/rust/catcollar/mod.rs

```diff
@@ -155,45 +156,31 @@ impl CatcollarLibOS {
     /// Binds a socket to a local endpoint.
     pub fn bind(&mut self, qd: QDesc, local: SocketAddrV4) -> Result<(), Fail> {
         trace!("bind() qd={:?}, local={:?}", qd, local);
-        let mut qtable: RefMut<IoQueueTable<CatcollarQueue>> = self.qtable.borrow_mut();
         // Check if we are binding to the wildcard port.
         if local.port() == 0 {
             let cause: String = format!("cannot bind to port 0 (qd={:?})", qd);
             error!("bind(): {}", cause);
             return Err(Fail::new(libc::ENOTSUP, &cause));
         }
 
-        // Check if queue descriptor is valid.
-        if qtable.get(&qd).is_none() {
-            let cause: String = format!("invalid queue descriptor {:?}", qd);
-            error!("bind(): {}", &cause);
-            return Err(Fail::new(libc::EBADF, &cause));
-        }
-
-        // Check wether the address is in use.
-        for (_, queue) in qtable.get_values() {
-            if let Some(addr) = queue.get_addr() {
-                if addr == local {
-                    let cause: String = format!("address is already bound to a socket (qd={:?}", qd);
-                    error!("bind(): {}", &cause);
-                    return Err(Fail::new(libc::EADDRINUSE, &cause));
-                }
-            }
+        // Check whether the address is in use.
+        if self.addr_in_use(local) {
+            let cause: String = format!("address is already bound to a socket (qd={:?}", qd);
+            error!("bind(): {}", cause);
+            return Err(Fail::new(libc::EADDRINUSE, &cause));
         }
-
-        // Get a mutable reference to the queue table.
-        // It is safe to unwrap because we checked before that the queue descriptor is valid.
-        let queue: &mut CatcollarQueue = qtable.get_mut(&qd).expect("queue descriptor should be in queue table");
-
         // Get reference to the underlying file descriptor.
-        // It is safe to unwrap because when creating a queue we assigned it a valid file descritor.
-        let fd: RawFd = queue.get_fd().expect("queue should have a file descriptor");
+        let fd: RawFd = self.get_queue_fd(&qd)?;
 
         // Bind underlying socket.
         let saddr: SockAddr = linux::socketaddrv4_to_sockaddr(&local);
         match unsafe { libc::bind(fd, &saddr as *const SockAddr, mem::size_of::<SockAddrIn>() as Socklen) } {
             stats if stats == 0 => {
-                queue.set_addr(local);
+                self.qtable
+                    .borrow_mut()
+                    .get_mut::<CatcollarQueue>(&qd)
+                    .expect("queue should exist")
+                    .set_addr(local);
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?